ndk/src/media/media_codec.rs

```diff
@@ -229,14 +229,14 @@ impl MediaCodec {
     pub fn configure(
         &self,
         format: &MediaFormat,
-        surface: &NativeWindow,
+        surface: Option<&NativeWindow>,
         direction: MediaCodecDirection,
     ) -> Result<()> {
         let status = unsafe {
             ffi::AMediaCodec_configure(
                 self.as_ptr(),
                 format.as_ptr(),
-                surface.ptr().as_ptr(),
+                surface.map_or(ptr::null_mut(), |s| s.ptr().as_ptr()),
                 ptr::null_mut(),
                 if direction == MediaCodecDirection::Encoder {
                     1
```

Comment:
While at it, can you replace this with:
```suggestion
                    ffi::AMEDIACODEC_CONFIGURE_FLAG_ENCODE
```
No extra PR needed for that.